src/Hosting/Systemd/IntegrationTestApp/IntegrationTestApp.csproj

```diff
@@ -0,0 +1,13 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
```

Comment:
Move to src/Hosting/Systemd/testassets/

Reply:
@Tratcher I'm trying to find out how I can build and run this app so it uses libraries that are part of this repo. Is there an application in the repo already that I can have a look at?

Reply:
Our location conventions matter a lot. Start by moving it to a testassets folder.
https://github.com/aspnet/AspNetCore/blob/master/src/Servers/testassets/ServerComparison.TestSites/ServerComparison.TestSites.csproj

Reply:
I've moved the app to the testassets folder.
It ends up being built at `./artifacts/bin/IntegrationTestApp/Debug/netcoreapp3.0/IntegrationTestApp.dll`.
Trying to run that doesn't use the proper libraries:
```
INVOCATION_ID=not-empty .dotnet/dotnet ./artifacts/bin/IntegrationTestApp/Debug/netcoreapp3.0/IntegrationTestApp.dll

Unhandled Exception: System.MissingMethodException: Method not found: 'Void Microsoft.Extensions.Logging.Console.ConsoleLoggerOptions.set_Format(Microsoft.Extensions.Logging.Console.ConsoleLoggerFormat)'.
   at Microsoft.Extensions.Hosting.SystemdHostBuilderExtensions.<>c.<UseSystemd>b__0_1(ConsoleLoggerOptions options)
   at Microsoft.Extensions.Options.ConfigureNamedOptions`1.Configure(String name, TOptions options)
   at Microsoft.Extensions.Options.OptionsFactory`1.Create(String name)
   at Microsoft.Extensions.Options.OptionsMonitor`1.<>c__DisplayClass11_0.<Get>b__0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at Microsoft.Extensions.Options.OptionsCache`1.GetOrAdd(String name, Func`1 createOptions)
...
```

Reply:
@aspnet/build ?

Reply:
It might be problematic to be running tests against AspNetCore APIs in this repo. The dependency flow of our repos is dotnet/core-setup -> Extensions -> AspNetCore -> core-sdk. To avoid a cycle, this repo is using an LKG version of the SDK and AspNetCore runtime. If you were to need a new API from or need to make a change that impacts AspNetCore, you would have to wait for the next stable or preview release to cycle that change back down to Extensions.

The other problem you might run into is the way the SDK trims assemblies against a FrameworkReference. I'm not 100% sure if that is what causes the missing method error right now, but this is where I would start looking. When the SDK sees the same assembly name in a project/package reference and a FrameworkReference, it chooses the higher assembly version or file version. It's probable that at some point this will be a problem for this test because local builds of this repo might be producing assemblies of a lower version than what is in the shared framework.

So, recommendation...consider moving tests that rely on AspNetCore into the AspNetCore repo. Also, binary logs are a great way to investigate issues related to conflict resolution of assemblies.

Reply:
The test doesn't rely on AspNetCore... Is our mistake just using Microsoft.NET.Sdk.Web? I found one other test app in this repo and it uses Microsoft.Net.Sdk.
https://github.com/aspnet/Extensions/blob/master/src/Hosting/test/testassets/Microsoft.Extensions.Hosting.TestApp/Microsoft.Extensions.Hosting.TestApp.csproj

Reply:
Oh, I missed the part with ConfigureWebHostDefaults and Startup. Those can be removed.

Reply:
> Is our mistake just using Microsoft.NET.Sdk.Web?

:+1:, looking better!
```
 INVOCATION_ID=notnull .dotnet/dotnet  ./artifacts/bin/IntegrationTestApp/Debug/netcoreapp3.0/IntegrationTestApp.dll
<6>Microsoft.Hosting.Lifetime[0] Application started. Hosting environment: Production; Content root path: /home/tmds/repos/AspNetExtensions
```